nengo_spa/algebras/vtb_algebra.py

```diff
@@ -155,6 +218,81 @@ def invert(self, v, sidedness=ElementSidedness.TWO_SIDED):
         sub_d = self._get_sub_d(len(v))
         return v.reshape((sub_d, sub_d)).T.flatten()
 
+    def binding_power(self, v, exponent):
+        r"""Returns the binding power of *v* using the *exponent*.
+
+        The binding power is defined as binding (*exponent*-1) times bindings
+        of *v* to itself.
+
+        Fractional binding powers are supported for "positive" vectors if SciPy
+        is available.
+
+        Note the following special exponents:
+
+        * an exponent of -1 will return the inverse,
+        * an exponent of 0 will return the identity vector,
+        * and an *exponent* of w1cne will return *v* itself.
+
+        Be aware that the binding power for the VTB algebra does *not* satisfy
+        the usual properties of exponentiation:
+
+        * :math:`\mathcal{B}(v^a, v^b) = v^{a+b}` does *not* hold,
+        * :math:`(v^a)^b = v^{ab}` does *not* hold.
+
+        Parameters
+        ----------
+        v : (d,) ndarray
+            Vector to bind repeatedly to itself.
+        exponent : int or float
+            Exponent of the binding power.
+
+        Returns
+        -------
+        (d,) ndarray
+            Binding power of *v*.
+
+        See also
+        --------
+        .sign
+        """
+
+        try:
+            from scipy.linalg import fractional_matrix_power
+        except ImportError:
+            if int(exponent) == exponent:
+                exponent = int(exponent)
+                # Provide fallback for integer-only powers
+
+                def fractional_matrix_power(m, exp):
+                    power = np.eye(len(m))
+                    for _ in range(exp):
+                        power = np.dot(power, m)
+                    return power
```

Comment:
I think there should be a way to write this so that it uses the base class implementation as a fallback? Also note these lines are uncovered (and same with the base class implementation).

Reply:
Pushed a commit to turn codecov on for the `no-scipy` job. Nengo appears to do the same thing. Currently we're seeing codecov patch report `84.32% of diff hit (target 90.00%)` and codecov project report `95.49% (target 90.00%)` so I'll see what it says afterward. It should go up due to `test_algebra.py::test_integer_binding_power` running without scipy, triggering the `ImportError` and then running these blocks.

Reply:
codecov/patch: `90.29% of diff hit (target 90.00%)`
codecov/project `96.15% (target 90.00%)`

And these lines are now covered, so looks like that worked.

Remaining question from this thread:
> I think there should be a way to write this so that it uses the base class implementation instead of this fallback?

Reply:
I can see the desire to reduce the code duplication, but a few points kept me from trying to hard in this instance:

* It is not entirely clear to me that we have actual semantic code duplication and not just syntactic duplication. In a sense VTB and TVTB should be handled as independent algebras. (Even though they are based on the same idea, which that might suggest that there might be a reasonable common basis to extract.)
* Extracting duplicated code adds some indirection/more complex code which might make it harder to understand each of these algebras.
* [Rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming))